libffi-sys-rs/build/not_msvc.rs

```diff
@@ -50,8 +52,28 @@ pub fn configure_libffi(prefix: PathBuf, build_dir: &Path) {
     command
         .arg("configure")
         .arg("--with-pic")
-        .arg("--disable-docs")
-        .current_dir(&build_dir);
+        .arg("--disable-docs");
+
+    // Enable debug assertions in libffi when targeting debug in cargo
+    // NOTE: `--enable-debug` doesn't build right for older versions of the x86_64-pc-windows-gnu
+    // toolchain, so we don't enable debug assertions prior to 1.47.0
+
+    let is_x86_64_pc_windows_gnu =
+        Ok("x86_64-pc-windows-gnu") != std::env::var("TARGET").as_ref().map(|s| s.as_str());
+    let is_pre_1_47_0 = rustc_version::version().map_or(false, |v| v < Version::new(1, 47, 0));
+
+    if !(is_x86_64_pc_windows_gnu && is_pre_1_47_0) {
+        if Ok("debug") == std::env::var("PROFILE").as_ref().map(|s| s.as_str()) {
+            command.arg("--enable-debug");
+        }
+    }
+
+    let target = std::env::var("TARGET").unwrap();
+    if target != std::env::var("HOST").unwrap() {
+        command.arg(format!("--host={}", target.to_string()));
+    }
+
+    command.current_dir(&build_dir);
```

Comment:
Are these changes really necessary? I'd rather not introduce additional dependencies _just_ so we can enable debug assertions.

Reply:
No, the debug assertions aren't necessary. They DID catch a bug and prevented undefined behavior when targeting `powerpc-unknown-linux-gnu`, though I haven't been able to fix the bug yet.

I can remove if you'd like.

Reply:
Since it's not directly related to this PR I'd prefer to not include it. I'm fine with including it in a separate PR, but it would be nice if we could do so without adding any additional (build time) dependencies.